crates/index-scheduler/src/batch.rs

```diff
@@ -1805,6 +1899,7 @@ impl IndexScheduler {
                     }
                 }
             }
+            atomic_progress.fetch_add(1, Ordering::Relaxed);
```

Comment:
consider inspecting the iterator to perform the fetch add while iterating (more robust in the face of early continues)

Reply:
In the end we said it wasn't a good idea since